src/sinks/gcp/chronicle_unstructured.rs

```diff
@@ -0,0 +1,522 @@
+                encoder.encode(event, &mut bytes).ok()?;
+
+                let mut value = json!({
+                    "log_text": String::from_utf8_lossy(&bytes),
+                });
+
+                if let Some(ts) = timestamp {
+                    value.as_object_mut().unwrap().insert(
+                        "ts_rfc3339".to_string(),
+                        ts.to_rfc3339_opts(chrono::SecondsFormat::AutoSi, true)
+                            .into(),
+                    );
+                }
+
+                Some(value)
+            })
+            .collect::<Vec<_>>();
+
+        let json = json!({
+            "customer_id": self.customer_id,
+            "log_type": partition_key,
+            "entries": events,
+        });
+
+        let body = crate::serde::json::to_bytes(&json)?.freeze();
+        writer.write_all(&body)?;
+        Ok(body.len())
+    }
+}
+
+// Settings required to produce a request that do not change per
+// request. All possible values are pre-computed for direct use in
+// producing a request.
+#[derive(Clone, Debug)]
+struct RequestSettings {
+    encoder: ChronicleEncoder,
+
+    // TODO Does chronicle handle compression?
+    compression: Compression,
+}
+
+struct ChronicleRequestPayload {
+    bytes: Bytes,
+}
+
+impl From<Bytes> for ChronicleRequestPayload {
+    fn from(bytes: Bytes) -> Self {
+        Self { bytes }
+    }
+}
+
+impl AsRef<[u8]> for ChronicleRequestPayload {
+    fn as_ref(&self) -> &[u8] {
+        self.bytes.as_ref()
+    }
+}
+
+impl RequestBuilder<(String, Vec<Event>)> for RequestSettings {
+    type Metadata = (EventFinalizers, RequestMetadataBuilder);
+    type Events = (String, Vec<Event>);
+    type Encoder = ChronicleEncoder;
+    type Payload = ChronicleRequestPayload;
+    type Request = ChronicleRequest;
+    type Error = io::Error;
+
+    fn compression(&self) -> Compression {
+        self.compression
+    }
+
+    fn encoder(&self) -> &Self::Encoder {
+        &self.encoder
+    }
+
+    fn split_input(&self, input: (String, Vec<Event>)) -> (Self::Metadata, Self::Events) {
+        let (partition_key, mut events) = input;
+        let finalizers = events.take_finalizers();
+
+        let metadata = RequestMetadata::builder(&events);
+        ((finalizers, metadata), (partition_key, events))
+    }
+
+    fn build_request(
+        &self,
+        metadata: Self::Metadata,
+        payload: EncodeResult<Self::Payload>,
+    ) -> Self::Request {
+        let (finalizers, metadata_builder) = metadata;
+
+        let metadata = metadata_builder.build(&payload);
+        let body = payload.into_payload().bytes;
+
+        ChronicleRequest {
+            body,
+            finalizers,
+            metadata,
+        }
+    }
+}
+
+impl RequestSettings {
+    fn new(config: &ChronicleUnstructuredConfig) -> crate::Result<Self> {
+        let transformer = config.encoding.transformer();
+        let (_, serializer) = config.encoding.config();
```

Comment:
If we don't need the framer, we use the `EncodingConfig` struct instead of `EncodingConfigWithFraming`